src/formatDate.js

```diff
@@ -8,7 +8,42 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let dateArr = [];
+  let resultString = '';
+  let index = 0;
+
+  dateArr = date.split(fromFormat[3]);
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    const fromIndex = fromFormat.indexOf(toFormat[i]);
+
+    index++;
+
+    if (toFormat[i] === 'YY' && fromFormat.includes('YYYY')) {
+      const number = dateArr[index - 1];
+
+      resultString += number.slice(-2);
+    } else if (toFormat[i] === 'YYYY' && fromFormat.includes('YY')) {
+      const year = Number(dateArr[index - 1]);
+
+      if (dateArr[index - 1] === '00') {
+        resultString += '2000.';
+        continue;
+      } else {
+        const fullYear = year < 30 ? `20${year}` : `19${year}`;
+
+        resultString += fullYear;
+      }
+    } else {
+      resultString += dateArr[fromIndex];
+    }
+
+    if (i < toFormat.length - 2) {
+      resultString += toFormat[3];
+    }
```

Comment:
This condition adds the separator to the `resultString`. However, it does not check if the current element of `toFormat` is a part of the date or the separator. This can lead to incorrect results if `toFormat` contains the separator at a different position than `fromFormat`.